.github/workflows/build-and-push-assets.yml

```
@@ -290,7 +280,7 @@ jobs:
           git commit -m "[BOT] Add compiled assets for #${{ github.ref }}" --no-verify || ((echo "NO_CHANGES=yes" >> $GITHUB_ENV) && (echo "No changes to commit"))
 
       - name: Git push for branch
-        if: ${{ github.ref_type == 'branch' && (env.NO_CHANGES != 'yes' || inputs.BUILT_BRANCH_SUFFIX != '' || inputs.BUILT_BRANCH_NAME != '') }}
+        if: ${{ github.ref_type == 'branch' && (env.NO_CHANGES != 'yes' || inputs.BUILT_BRANCH_NAME != '') }}
```

Shouldn't `BUILT_BRANCH_NAME` be read from env since, in line 205, we fall back to `GitHub.ref_name`?

We might end in an infinite workflow run loop; see: 
Someone pushes to dev-main, and `BUILT_BRANCH_NAME` is not defined; the workflow runs and pushes the build to the same branch due to the assignment in 205.

Do we want to allow then an empty `BUILT_BRANCH_NAME` to continue with the workflow to push to the same branch?

```suggestion
        if: ${{ github.ref_type == 'branch' && (env.NO_CHANGES != 'yes' || env.BUILT_BRANCH_NAME != '') }}
```

---

we decided to leave it as it is, I did a test without the change and there is no infinite loop happening